fix(DB/Core): Dragonmaw Peon AI

Stoabrogga opened the pull request:
##### CHANGES PROPOSED:
This PR contains basically the changes from #1676 and #1678 with the following modifications:
- grant the kill credit before the "Vomit" (43327) spell is applied, otherwise the player does not get the quest credit (I think this was what pklloveyou meant in his review of PR #1676)
- added another suggestion from pklloveyou to force the peons facing the mutton before eating and despawning the mutton after finishing eating (looks better in my opinion)
- leave the variable "Tapped" in the code (was removed in the other PRs); it is used to avoid an already vomiting peon to run to a new mutton, which would otherwise look strange
- add all broadcast text IDs to the creature text SQL script
- add an additional creature text for the poisoned peon

###### ISSUES ADDRESSED:
Closes #1707 

##### TESTS PERFORMED:
tested build (Ubuntu 16.04, clang 7) and in-game

##### HOW TO TEST THE CHANGES:
This is a little bit complicated. It is not enough to just add the quest "A Slow Death" (11020), as @comix1988 correctly stated in #1676. Instead to prepare for this as GM you have to add the whole quest chain leading to the daily quest 11020. Please test as follows (with GM mode off!):

- remove all associated quests of the quest chain and get hated with the netherwing:
```
.quest remove 10870
.quest remove 11012
.quest remove 11013
.quest remove 11014
.quest remove 11019
.quest remove 11020
.modify reputation 1015 hated
.go -4862.29 410.73 61.0058 530
```
- kill the Dragonmaw Peons and check that they are using their new abilities "Kick" and "Sunder Armor"
- add and reward all associated quests of the quest chain:
```
.quest add 10870
.quest reward 10870
.quest add 11012
.quest reward 11012
.quest add 11013
.quest reward 11013
.quest add 11014
.quest reward 11014
.quest add 11019
.quest reward 11019
.quest add 11020
.additem 32502 12
```
- proceed with the quest "A Slow Death"

##### KNOWN ISSUES AND TODO LIST:
none

##### Target branch(es):
Master

Pondaveia commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -556,6 +556,24 @@ class npc_enslaved_netherwing_drake : public CreatureScript
 /*#####
 # npc_dragonmaw_peon
 #####*/
+enum DragonmawPeon
```
codestyle: align all the equals for readability

Pondaveia commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -571,64 +589,112 @@ class npc_dragonmaw_peon : public CreatureScript
     {
         npc_dragonmaw_peonAI(Creature* creature) : ScriptedAI(creature) { }
 
+        EventMap events;
```
codestyle: EventMap events, events2;

Pondaveia commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -571,64 +589,112 @@ class npc_dragonmaw_peon : public CreatureScript
-            PoisonTimer = 0;
+        }
+
+        void EnterCombat(Unit* /*who*/)
```
Don't forget to override functions :)

Pondaveia commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -571,64 +589,112 @@ class npc_dragonmaw_peon : public CreatureScript
-                me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_EAT);
-                PoisonTimer = 15000;
+                Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID);
+                if (player && player->GetQuestStatus(11020) == QUEST_STATUS_INCOMPLETE)
```
Maybe create a constant for these magic numbers. Also the one line below.

Pondaveia commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -571,64 +589,112 @@ class npc_dragonmaw_peon : public CreatureScript
         void UpdateAI(uint32 diff)
         {
-            if (PoisonTimer)
+            events.Update(diff);
```
Just for notice, you can have the same eventmap for both Out and in combat. As long as the checks out of combat are inside !updatevictim() they will never be triggered. So you can get rid of 1 of the events variable.

Pondaveia commented on src/server/scripts/Outland/zone_shadowmoon_valley.cpp:
```diff
@@ -571,64 +589,112 @@ class npc_dragonmaw_peon : public CreatureScript
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_CHEER);
+                    break;
+                case EVENT_POISONED:
+                    if (GameObject* food = me->FindNearestGameObject(185893, 5.0f))
```
Constant for magic numbers

Meltie2013 commented on data/sql/updates/pending_db_world/rev_1555327481238291012.sql:
```diff
@@ -0,0 +1,11 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1555327481238291012');
+
+DELETE FROM `creature_text` WHERE `creatureid` = '22252';
```
Creature Id doesn't require ' ' for the entries, but can be still used.

Stoabrogga commented:
@Pondaveia @Meltie2013 
Thanks for your feedback. I updated the PR accordingly and also added an additional creature text for the poisoned peon which was missing.

Pondaveia commented:
Tested and only found this issue:
When you put the food down, the peons will attack the player when they get into aggro range

Stoabrogga commented:
@Pondaveia 
It works for me, they are not attacking me when they are friendly. Are you sure you included all steps while testing? I tested like this:

1. Reset quest chain and reputation with the Netherwing, teleport to Netherwing Ledge:
```
.quest remove 10870
.quest remove 11012
.quest remove 11013
.quest remove 11014
.quest remove 11019
.quest remove 11020
.modify reputation 1015 hated
.go -4862.29 410.73 61.0058 530
```
As you don't went through the quest chain all Dragonmaw Peons are hostile and will attack on sight (this is for testing if they are using the correct abilities).

2. Get all necessary quests of the quest chain up to the daily quest "A Slow Death" and provide the item "Fel Gland":
```
.quest add 10870
.quest reward 10870
.quest add 11012
.quest reward 11012
.quest add 11013
.quest reward 11013
.quest add 11014
.quest reward 11014
.quest add 11019
.quest reward 11019
.quest add 11020
.additem 32502 12
```
You automatically get the Fel Orc disguise aura and all Dragonmaw Peons are now friendly. Place the mutton before them and they will just eat it and die. That's at least what happens for me. They are not attacking.

Stoabrogga commented:
Here's a small video showing how it looks on my server (only part 2: poisoning the peons):
[dragonmaw_peon.zip](https://github.com/azerothcore/azerothcore-wotlk/files/3083324/dragonmaw_peon.zip)

Pondaveia commented:
I thought i wasn't supposed to have the aura and i removed it, in this case i think the PR is good
